src/NzbDrone.Core/Notifications/Pushsafer/PushsaferSettings.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+using FluentValidation;
+using NzbDrone.Core.Annotations;
+using NzbDrone.Core.ThingiProvider;
+using NzbDrone.Core.Validation;
+
+namespace NzbDrone.Core.Notifications.Pushsafer
+{
+    public class PushsaferSettingsValidator : AbstractValidator<PushsaferSettings>
+    {
+        public PushsaferSettingsValidator()
+        {
+            RuleFor(c => c.ApiKey).NotEmpty();
+            RuleFor(c => c.Retry).GreaterThanOrEqualTo(60).LessThanOrEqualTo(10800).When(c => (PushsaferPriority)c.Priority == PushsaferPriority.Emergency);
+            RuleFor(c => c.Expire).GreaterThanOrEqualTo(60).LessThanOrEqualTo(10800).When(c => (PushsaferPriority)c.Priority == PushsaferPriority.Emergency);
+        }
+    }
+
+    public class PushsaferSettings : IProviderConfig
+    {
+        private static readonly PushsaferSettingsValidator Validator = new ();
+
+        public PushsaferSettings()
+        {
+            Priority = 0;
+            DeviceIds = Array.Empty<string>();
+        }
+
+        [FieldDefinition(0, Label = "API Key", Privacy = PrivacyLevel.ApiKey, HelpLink = "https://www.pushsafer.com/en/pushapi_ext#API-K")]
+        public string ApiKey { get; set; }
+
+        [FieldDefinition(1, Label = "Device IDs", HelpText = "Device Group ID or list of Device IDs (leave blank to send to all devices)", Type = FieldType.Tag, Placeholder = "123456789|987654321")]
+        public IEnumerable<string> DeviceIds { get; set; }
+
+        [FieldDefinition(2, Label = "Priority", Type = FieldType.Select, SelectOptions = typeof(PushsaferPriority))]
+        public int Priority { get; set; }
+
+        [FieldDefinition(3, Label = "Retry", Type = FieldType.Textbox, HelpText = "Interval to retry Emergency alerts, minimum 60 seconds")]
+        public int Retry { get; set; }
+
+        [FieldDefinition(4, Label = "Expire", Type = FieldType.Textbox, HelpText = "Maximum time to retry Emergency alerts, maximum 10800 seconds")]
```

Comment:
Should some of these be advanced settings?

Reply:
Makes sense, will do.

Reply:
Makes sense to leave those two as normal in that case, how about the others

Reply:
Honestly not sure which ones we could consider as "advanced". Maybe device IDs or any of the optional ones like sound, vibration, icon and icon color.

Reply:
`sound, vibration, icon and icon color`

These were the ones I was thinking